### PR TITLE
refactor: driver validation, documents types changes and console code removed

### PR DESCRIPTION
**Pull request by Daljit-fleetHawks**
## Change Description

I have resolved validation issue on driver form and updated documents types in driver list. and I did removed console.log code from some components like driver, geofences, maintenance etc. Also resolved routing issue on header links.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist: (Check only applicable)

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/pages/fleet/drivers/driver-list/driver-list.component.ts [hunk 1]
```diff
@@ -330,13 +330,15 @@ export class DriverListComponent implements AfterViewInit, OnDestroy, OnInit {
...
-        .getData(`drivers/isDeleted/${driverID}/${value}`)
+        .getData(`drivers/isDeleted/${driverID}/${item.isDeleted}`)
         .subscribe((result: any) => {
-          this.rerender();
-          console.log('result', result);
+          this.toastr.success('Driver deleted successfully!');
+          this.drivers = this.drivers.filter(u => u.driverID !== item.driverID);
+          this.fetchDrivers();
         }, err => {
           console.log('driver delete', err);
```
Comment (amritfleethawks) [inline src/app/pages/fleet/drivers/driver-list/driver-list.component.ts]: please remove console log

File: src/app/pages/fleet/maintenance/service-log/service-detail/service-detail.component.ts [hunk 2]
```diff
@@ -104,6 +104,7 @@ export class ServiceDetailComponent implements OnInit {
     this.apiService.getData('vehicles/get/list')
       .subscribe((result: any) => {
         this.vehiclesObject = result;
+        console.log('vehiclesObject', this.vehiclesObject);
```
Comment (amritfleethawks) [inline src/app/pages/fleet/maintenance/service-log/service-detail/service-detail.component.ts]: please remove console log

File: src/app/pages/fleet/drivers/driver-list/driver-list.component.ts (hunk 1 above)
Comment (amritfleethawks) [inline src/app/pages/fleet/drivers/driver-list/driver-list.component.ts]: please remove console log

**Review by amritfleethawks — APPROVED** [review]

File: src/app/pages/fleet/assets/asset-detail/asset-detail.component.ts [hunk 3]
```diff
@@ -166,7 +165,6 @@ export class AssetDetailComponent implements OnInit {
       .subscribe((result: any) => {
         if (result) {
           this.deviceData = result['Items'];
-          console.log('deviceData', this.deviceData);
         }
       }, (err) => {
         console.log('asset detail', err);
```
Comment (kunalfleethawks) [inline src/app/pages/fleet/assets/asset-detail/asset-detail.component.ts]: Remove console.logs

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]
@Daljit-fleetHawks please resolve the comments

**Review by kunalfleethawks — APPROVED** [review]

**Review by Daljit-fleetHawks — COMMENTED** [review]
changes done

**Pull request merged**